src/Periphery/GasZip.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { TransferrableOwnership } from "../Helpers/TransferrableOwnership.sol";
+
+interface IGasZip {
+    function deposit(
+        uint256 destinationChain,
+        address recipient
+    ) external payable;
+}
+
+/// @title GasZip
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality to swap and trigger gaz.zip protocol
+/// @custom:version 1.0.0
+contract GasZip is TransferrableOwnership {
+    address public immutable ZERO = address(0);
+
+    /// State ///
+
+    mapping(address => bool) public allowedInboundTokens;
+    IGasZip public immutable gasZipRouter;
+
+    /// Errors ///
+    error SwapFailed(address, address);
+    error GasZipFailed(uint256);
+    error TransferFailed();
+    error InboundTokenDisallowed();
+
+    /// Events ///
+
+    /// Constructor ///
+
+    modifier inboundTokenIsAllowed(address token) {
+        if (!allowedInboundTokens[token]) revert InboundTokenDisallowed();
+        _;
+    }
+
+    constructor(
+        address _owner,
+        address _gasZipRouter
+    ) TransferrableOwnership(_owner) {
+        gasZipRouter = IGasZip(_gasZipRouter);
```

Comment:
should check that address is not `ZERO`

Reply:
our deploy scripts will check that the parameter is available and not a zero address. 
But for this the file ```deployRequirements.json``` need to be updated respectively

Reply:
@0xDEnYO updated deployRequirements. Please check 🙂
Also, please advice if we need to define it in DeployGasZip, or is it enough to have both `_owner` and `_gasZipRouter` defined in deployRequirements?
<img width="440" alt="image" src="https://github.com/lifinance/contracts/assets/46458690/f5ba5826-d3ac-4019-a77e-74d86ab762f5">

Reply:
Only the `gasZipRouters`. You can remove all the owner handling because the inboundToken handling is no longer needed.